packages/ksf-login/src/KSF/Login/Component.purs

```diff
@@ -129,10 +129,12 @@ receiveProps { props, state, setState, isFirstMount } = when isFirstMount do
              { callback_failure: mkEffectFn1 \a -> do
                 Console.log "Janrain SSO failure"
              , callback_success: mkEffectFn1 \a -> do
+                JanrainSSO.setSsoSuccess
                 Console.log "Janrain SSO success"
              , capture_error: mkEffectFn1 \a -> do
                 Console.log "Janrain SSO capture error"
              , capture_success: mkEffectFn1 \r@({ result: { accessToken, userData: { uuid } } }) -> do
+                JanrainSSO.setSsoSuccess
```

Comment:
Do we need to set it in both of them?

Reply:
`capture_success` is called when we call `check_session` and it captures the user's session left by the other site. `callback_success` is called when we call `set_session` (yes, it seems like `check_session` saves that callback somewhere so it can be called by `set_session`), weird, but it seems to work.

Reply:
The way I understand it works:

- We call `SSO.check_session { callback_success, capture_success }`
- If SSO fires, `capture_success` is executed with all needed info
- otherwise, the user provides credentials to Persona and it gives back the SSO code
- We call `SSO.set_session(ssoCode) which executes `callback_success`

This seemed as a strange flow at first, but I think I now understand the reasoning behind it. You can register all your SSO-related callbacks in one place (when calling `check_session`) which is very useful if they need to share some context. And then drive the thing with non-blocking calls like `set_session` and `end_session` from other places.